src/home/celebration/NftCelebration.tsx

```diff
@@ -0,0 +1,233 @@
+  const [showConfetti, setShowConfetti] = useState(false)
+  const confettiStartTime = useRef(0)
+
+  const canShowNftCelebration = useSelector(showNftCelebrationSelector)
+  const celebratedNft = useSelector(celebratedNftSelector)
+
+  const nfts = useSelector(nftsWithMetadataSelector)
+  const matchingNft = nfts.find(
+    (nft) =>
+      !!celebratedNft &&
+      !!celebratedNft.networkId &&
+      celebratedNft.networkId === nft.networkId &&
+      !!celebratedNft.contractAddress &&
+      celebratedNft.contractAddress === nft.contractAddress
+  )
+
+  const isVisible = canShowNftCelebration && matchingNft
+
+  const renderBackdrop = useCallback(
+    (props: BottomSheetDefaultBackdropProps) => (
+      <BottomSheetBackdrop {...props} disappearsOnIndex={-1} appearsOnIndex={0} />
+    ),
+    []
+  )
+
+  const renderHandleWithImage = useCallback(
+    () => (
+      <View style={styles.handleWithImage}>
+        {matchingNft && (
+          <NftMedia
+            shouldAutoScaleHeight
+            nft={matchingNft}
+            ErrorComponent={
+              <View style={styles.imageError}>
+                <ImageErrorIcon />
+              </View>
+            }
+            origin={NftOrigin.NftCelebration}
+            mediaType="image"
+          />
+        )}
+        <View style={styles.handleBar} />
+      </View>
+    ),
+    [matchingNft]
+  )
+
+  useEffect(() => {
+    if (isVisible) {
+      // Wait for the home screen to have less ongoing async tasks.
+      // This should help the bottom sheet animation run smoothly.
+      const timeoutId = setTimeout(() => {
+        bottomSheetRef.current?.expand()
+      }, 1000)
+      return () => clearTimeout(timeoutId)
+    }
+  }, [isVisible])
+
+  const handleBottomSheetPositionChange = (index: number) => {
+    if (!matchingNft) {
+      return // this should never happen
+    }
+
+    if (index === -1) {
+      ValoraAnalytics.track(HomeEvents.nft_celebration_displayed, {
+        networkId: matchingNft.networkId,
+        contractAddress: matchingNft.contractAddress,
+      })
+
+      vibrateSuccess()
+
+      confettiStartTime.current = Date.now()
+      setShowConfetti(true)
+    }
+  }
+
+  const handleCtaPress = () => {
+    bottomSheetRef.current?.close()
+  }
+
+  const handleConfettiFinish = () => {
+    finishCelebration({ userInterrupted: false })
+  }
+
+  const handleConfettiDismiss = () => {
+    finishCelebration({ userInterrupted: true })
+  }
+
+  const finishCelebration = ({ userInterrupted }: { userInterrupted: boolean }) => {
+    if (!matchingNft) {
+      return // this should never happen
+    }
+
+    ValoraAnalytics.track(HomeEvents.nft_celebration_animation_displayed, {
+      userInterrupted,
+      durationInSeconds: Math.round((Date.now() - confettiStartTime.current) / 1000),
+    })
+
+    setShowConfetti(false)
+
+    dispatch(nftCelebrationDisplayed())
+  }
+
+  if (!isVisible) {
+    return null
+  }
+
+  return (
+    <>
+      <GorhomBottomSheet
```

Comment:
nit: any chance we can use `BottomSheetBase` here? it could save some code (the background would come for free) - i think it'd work if you add an optional `handleComponent` prop to the `BottomSheetBase`? (the `onChange` handler would probably work with the `onClose` prop in `BottomSheetBase`)

Reply:
yes, that makes sense. in addition to `handleComponent` I've also forwarded `backgroundStyle` and `onChange` props:
* `backgroundStyle` because we have to tweak the default background to not interfere with the rounded corners of the image (default background doesn't hide overflow)
* `onChange` because the `onClose` event is dispatched not only when the bottom sheet index changes to `-1`, but also when the bottom sheet component unmounts, which caused analytics to be sent twice.
also, it caused a strange behavior on the Android emulator when sometimes `onClose` was fired even before the bottom sheet appeared. i guess the bottom sheet was mounted and unmounted at least once (this never happened on iOS)
having noticed this, relying on `onChange` seems more safe